packages/editor/src/extensions/code-block/component.tsx

```diff
@@ -49,6 +50,23 @@ export function CodeblockComponent(
     (l) => l.filename === language || l.alias?.some((a) => a === language)
   );
 
+  const cachedLanguage =
+    config.get<(typeof Languages)[number]>("codeBlockLanguage");
+
+  useEffect(() => {
+    if (languageDefinition) {
+      config.set("codeBlockLanguage", languageDefinition);
+      return;
+    }
+
+    if (!languageDefinition && cachedLanguage && cachedLanguage.filename) {
+      updateAttributes(
+        { language: cachedLanguage.filename },
+        { addToHistory: false, preventUpdate: false }
+      );
+    }
+  }, []);
```

Comment:
We shouldn't do this here. This will run for all codeblocks even though we just want to set the language once when a new codeblock is added. It'll also cause a flash where a codeblock is plaintext and then switches to the last used language.

Instead let's do this in `code-block.ts` in the `parseHTML` function of the `language` attribute. Not sure if it'll work or not though so give it a try.

Reply:
Unfortunately, `useEffect` is called for all codeblocks when the page loads. Also, I wasn't able to notice a flash when codeblock is plaintext and then switches to the local storage language. But, there is a minor bug in this PR: user won't ever be able to go back to plaintext codeblock until they delete the local storage entry. I think we should add plaintext as a language in `languages.json`. What do you think?

> Instead let's do this in code-block.ts in the parseHTML function of the language attribute. Not sure if it'll work or not though so give it a try.

I tried in `parseHTML` of the `language` attribute. The issue here is the `parseHTML` doesn't get called when a new code block is added, so we can't save the code block language if user pasted a code block from VSCode for example. I'll continue trying with `parseHTML`.

Reply:
Thanks for the comemnt, I realized we don't need the `useEffect` at all. Please review the latest changes.

(plus I've added Plaintext in languages, let me know if we should keep this or else I can revert)